packages/web3torrent/src/clients/payment-channel-client.ts

```diff
@@ -280,31 +282,31 @@ export class PaymentChannelClient {
   }
 
   async closeChannel(channelId: string): Promise<ChannelState> {
-    logger.info(`Waiting for my turn to close channel ${channelId}`);
-    /*
-     * There are currently zero or one ongoing UpdateChannel RPCs.
-     * Since the torrent is paused, if there is one ongoing UpdateChannel RPC, it will soon resolve,
-     * and there will be no further UpdateChannel RPCs.
-     * If we don't wait, then a CloseChannel call could race an existing UpdateChannel call, and one of them
-     * would fail.
-     * Therefore, we wait 500ms before waiting for my turn.
-     */
-
-    await new Promise(resolve => setTimeout(resolve, 500));
-
-    /*
-     * Now, any UpdateChannel RPC should be finished, so when it's my turn, the CloseChannel
-     * call should succeed.
-     */
-
     const closing = this.channelState(channelId)
-      .pipe(first(cs => this.canUpdateChannel(cs)))
-      .subscribe(cs => {
+      .pipe(
+        filter(cs => this.canUpdateChannel(cs)),
+        take(5) // Limit to 5 attempts
+      )
+      .subscribe(async cs => {
         logger.info(
           {channelId, cs, me: this.mySigningAddress},
           "It's my turn, closing the channel"
         );
-        this.channelClient.closeChannel(channelId);
+        try {
+          await this.channelClient.closeChannel(channelId);
```

Comment:
In the case when the seeder is initiating the channel close, this call always race (payment acceptance) [https://github.com/statechannels/monorepo/blob/master/packages/web3torrent/src/library/web3torrent-lib.ts#L340]

Reply:
So, the torrent client should probably switch modes and not accept payments. Then, there would be at most once race.

Reply:
So, the torrent client should probably switch modes and not accept payments. Then, there would be at most once race.